.github/workflows/python.yml

```
@@ -30,25 +33,116 @@ jobs:
         environment-file: .github/python_test_env.yml
 
     - name: Install
-      run: pip install . -v
+      run: python -m pip install . -v
 
     - name: Test
-      run: pytest --cov exactextract --cov-report lcov python/tests
+      run: python -m pytest --cov exactextract --cov-report lcov python/tests
 
     - name: Upload coverage report
       uses: codecov/codecov-action@v3
       with:
         files: coverage.lcov
         token: ${{ secrets.CODECOV_TOKEN }}
-      
+
     - name: Build source distribution
       if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
       run: python -m build . --sdist
 
     - name: Upload distribution(s)
-      if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
+      if: startsWith(github.event.ref, 'refs/tags') || github.event_name == 'release'
```

```suggestion
      if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
```

I think it is fine to keep the original here, which ensures that the sdist is uploaded as github artifacts every time this build is run, and not only on releases (e.g. so you can actually check the generated sdist before doing a release)